internal/security/proxy/service.go

```diff
@@ -169,6 +191,102 @@ func (s *Service) Init(cp bootstrapConfig.CertKeyPair) error {
+// Name.URL to be considered well-formed
+// Name is used as the key of service name
+// URL should be well-formed as protocol://hostName:portNumber
+// and is parsed into bootstrapConfig.ClientInfo structure if valid
+// returns error if not valid
+func (s *Service) parseAdditionalProxyRoutes() (map[string]bootstrapConfig.ClientInfo, error) {
+	emptyMap := make(map[string]bootstrapConfig.ClientInfo)
+
+	routesFromEnv := strings.Split(s.additionalRoutes, ",")
+
+	additionalClientMap := make(map[string]bootstrapConfig.ClientInfo)
+	for _, rt := range routesFromEnv {
+		route := strings.TrimSpace(rt)
+		// ignore the empty route
+		if route == "" {
+			continue
+		}
+
+		if !strings.Contains(route, ".") {
+			// Invalid syntax for route, it should contain dot (.)
+			return emptyMap, fmt.Errorf(
+				"invalid syntax for defining additional kong route %s, it should contain dot . as separator", route)
+		}
+
+		routePair := strings.Split(route, ".")
+		serviceName := strings.TrimSpace(routePair[0])
+		routeURL := strings.TrimSpace(routePair[1])
+
+		if serviceName == "" {
+			// service name should not be empty
+			return emptyMap, errors.New("service name for kong route should not be empty")
+		}
+
+		// sanity check to validate the well-formness of routeURL
+		// and also parse out the protocol, hostname, and port number if it is good
+		url, err := url.Parse(routeURL)
+		if err != nil {
+			return emptyMap, fmt.Errorf(
+				"malformed route URL for additional kong route %s: %s", routeURL, err.Error())
+		}
+		hostName, port, err := net.SplitHostPort(url.Host)
+		if err != nil {
+			return emptyMap, fmt.Errorf(
+				"malformed host in route URL for additional kong route %s: %s", url.Host, err.Error())
+		}
+		portNum, err := strconv.Atoi(port)
+		if err != nil {
+			return emptyMap, fmt.Errorf(
+				"invalid port, expecting integer as port number for additional kong route %s: %s", port, err.Error())
+		}
+
+		clientInfo := bootstrapConfig.ClientInfo{
+			Protocol: url.Scheme,
+			Host:     hostName,
+			Port:     portNum,
+		}
+
+		additionalClientMap[serviceName] = clientInfo
+	}
+
+	return additionalClientMap, nil
+}
+
+func (s *Service) mergeMapWith(another map[string]bootstrapConfig.ClientInfo) map[string]bootstrapConfig.ClientInfo {
+	// merging ignroes the duplicate keys with the current internal map
+	// if the service name from another map already exists in the internal map of Servcie
+	// then that entry is ignored
+	if len(another) == 0 {
+		return s.configuration.Clients
+	}
+
+	if len(s.configuration.Clients) == 0 {
+		return another
+	}
+
+	merged := make(map[string]bootstrapConfig.ClientInfo)
+	for serviceName, client := range s.configuration.Clients {
+		merged[serviceName] = client
+	}
+
+	for serviceName, client := range another {
+		_, exists := merged[serviceName]
+		if exists {
+			s.loggingClient.Warn(fmt.Sprintf("service name %s already exists in the configuration", serviceName))
```

Comment:
tweak this to start as `"Attempting to add additional service name %s that already exists... Ignoring additional",...`

Reply:
sure